packages/build/src/plugins/compatibility.js

```diff
@@ -42,7 +42,12 @@ export const getExpectedVersion = async function ({ versions, nodeVersion, packa
 //  - Otherwise, use `latestVersion`
 const getCompatibleEntry = async function ({ versions, nodeVersion, packageJson, buildDir, pinnedVersion }) {
   if (pinnedVersion !== undefined) {
-    return versions.find(({ version }) => semver.satisfies(version, pinnedVersion)) || { version: pinnedVersion }
+    // invalid semver versions are coerced to valid ones (e.g 4.17.1-runtime.7 -> 4.17.1)
+    return (
+      versions.find(({ version }) => semver.satisfies(semver.coerce(version), pinnedVersion)) || {
```

Comment:
```suggestion
       versions.find(({ version }) => semver.satisfies(version, pinnedVersion, { includePrerelease: true })) || {
```

This [should work](https://runkit.com/embed/g9boqlslttmy)